packages/react-native/ReactCommon/react/renderer/core/RawProps.cpp

```diff
@@ -172,13 +172,38 @@ void RawProps::parse(const RawPropsParser& parser) noexcept {
  * The support for explicit conversion to `folly::dynamic` is deprecated and
  * will be removed as soon Android implementation does not need it.
  */
-RawProps::operator folly::dynamic() const noexcept {
+RawProps::operator folly::dynamic() const {
+  return toDynamic();
+}
+
+/*
+ * Deprecated. Do not use.
+ * The support for explicit conversion to `folly::dynamic` is deprecated and
+ * will be removed as soon Android implementation does not need it.
+ */
+folly::dynamic RawProps::toDynamic(
+    const std::function<bool(const std::string&)>& filterObjectKeys) const {
   switch (mode_) {
     case Mode::Empty:
       return folly::dynamic::object();
-    case Mode::JSI:
-      return jsi::dynamicFromValue(
-          *runtime_, value_, ignoreYogaStyleProps_ ? isYogaStyleProp : nullptr);
+    case Mode::JSI: {
+      const std::function<bool(const std::string&)> propFilterFunction =
+          [&](const std::string& key) {
+            if (ignoreYogaStyleProps_ && isYogaStyleProp(key)) {
+              return false;
+            }
+            if (filterObjectKeys) {
+              return filterObjectKeys(key);
+            }
+            return true;
+          };
```

Comment:
```suggestion
      const std::function<bool(const std::string&)> propFilterFunction =
          [ignoreYogaStyleProps = ignoreYogaStyleProps_, filterObjectKeys](const std::string& key) {
            if (ignoreYogaStyleProps && isYogaStyleProp(key)) {
              return false;
            }
            if (filterObjectKeys) {
              return filterObjectKeys(key);
            }
            return true;
          };
```
This function doesn't properly capture the other function.

Reply:
Nevermind, the `propFilterFunction` doesn't need to _capture_ `filterObjectKeys`, because `propFilterFunction` will always be executed within the scope of the parent function, so we never need to keep the ref for longer. `[&]` is fine here, I'd just maybe explain this